WordPress/gutenberg
# Button: Add opt-in prop for larger `isSmall` size

Part of #46741

## What?

Adds a `__next32pxSmallSize` prop to opt into the new 32px sizes when `isSmall=true`.

## Why?

To ensure a smooth transition from the current small size (24px) to the new small size (32px).

## Testing Instructions

1. `npm run storybook:dev`
2. For the "Primary" and "Icon" stories for Button, toggle the control for the `__next32pxSmallSize` prop. This should have no effect when `isSmall=false`.

## Screenshots or screencast 

Top row is `isSmall`, bottom row is `isSmall` & `__next32pxSmallSize`.

<img width="169" alt="The isSmall buttons in 24px and 32px sizes" src="https://github.com/WordPress/gutenberg/assets/555336/d18bffaf-4d40-4b0d-b330-99b77c0f1ccd">
